media-libs/zxing-cpp/zxing-cpp-2.2.1-r1.ebuild

```diff
@@ -0,0 +1,95 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+PYTHON_COMPAT=( python3_{10..13} )
+
+DISTUTILS_EXT=1
+DISTUTILS_OPTIONAL=1
+DISTUTILS_USE_PEP517=setuptools
+
+inherit cmake distutils-r1
+
+DESCRIPTION="C++ Multi-format 1D/2D barcode image processing library"
+HOMEPAGE="https://github.com/zxing-cpp/zxing-cpp"
+SRC_URI="https://github.com/${PN}/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz
+	test? ( https://github.com/${PN}/${PN}/releases/download/v${PV}/test_samples.tar.gz -> ${P}-test.tar.gz )
+"
+
+LICENSE="Apache-2.0"
+SLOT="0/3"
+KEYWORDS="~amd64 ~arm ~arm64 ~loong ~ppc64 ~riscv ~x86"
+IUSE="python test"
+
+DEPEND="python? ( dev-python/pybind11 )"
+RDEPEND="${PYTHON_DEPS}"
+BDEPEND="${DISTUTILS_DEPS}
+	test? (
+		dev-cpp/gtest
+		dev-libs/stb
+		dev-libs/libfmt
+	)
+"
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+RESTRICT="!test? ( test )"
+
+# https://github.com/zxing-cpp/zxing-cpp/issues/848
+CMAKE_WARN_UNUSED_CLI=no
+QA_PRESTRIPPED="usr/lib/python.*/site-packages/zxingcpp.cpython-.*.so"
+
+src_prepare() {
+	cmake_src_prepare
+
+	if use python; then
+		cd "${S}"/wrappers/python || die
+		distutils-r1_src_prepare
+	fi
+}
+
+src_configure() {
+	# TODO: Next release deprecates -DBUILD_* and replaces it with -DZXING_*
+	local mycmakeargs=(
+		-DBUILD_DEPENDENCIES=LOCAL
+
+		-DBUILD_EXAMPLES=OFF # nothing is installed
+
+		-DBUILD_BLACKBOX_TESTS=$(usex test ON OFF)
+		-DBUILD_UNIT_TESTS=$(usex test ON OFF)
+	)
+	cmake_src_configure
+
+	if use python; then
+		cd "${S}"/wrappers/python || die
+		# These would be nice to have, but since we're installing the library and python bindings
```

Comment:
This looks sus. Does it build the whole thing again? Or statically link into it? Perhaps the bindings really do belong in a separate package?

Reply:
I also considered if it would be better off as a separate standalone package.

I believe the barrier entry is higher for new packages than use flags. Would that be acceptable as a better approach?

Reply:
Also to answer the initial question. It builds the whole thing again for each python version.

Reply:
It would be a split of the existing package, so not really a problem for entry.